test/e2e/network/service.go

```diff
@@ -1852,8 +1853,112 @@ var _ = SIGDescribe("Services", func() {
 		}
 	})
 
+	// This test verifies if service load balancer cleanup finalizer can be removed
+	// when feature gate isn't enabled on the cluster.
+	// This ensures downgrading from higher version cluster will not break LoadBalancer
+	// type service.
+	ginkgo.It("should remove load balancer cleanup finalizer when service is deleted [Slow]", func() {
+		jig := framework.NewServiceTestJig(cs, "lb-remove-finalizer")
+
+		ginkgo.By("Create load balancer service")
+		svc := jig.CreateTCPServiceOrFail(f.Namespace.Name, func(svc *v1.Service) {
+			svc.Spec.Type = v1.ServiceTypeLoadBalancer
+		})
+
+		defer func() {
+			waitForServiceDeletedWithFinalizer(cs, svc.Namespace, svc.Name)
+		}()
+
+		ginkgo.By("Wait for load balancer to serve traffic")
+		svc = jig.WaitForLoadBalancerOrFail(svc.Namespace, svc.Name, framework.GetServiceLoadBalancerCreationTimeout(cs))
+
+		ginkgo.By("Manually add load balancer cleanup finalizer to service")
+		svc.Finalizers = append(svc.Finalizers, "service.kubernetes.io/load-balancer-cleanup")
```

Comment:
same here, use constant?

Reply:
Filed https://github.com/kubernetes/kubernetes/issues/78549 to track the followup.